protocol/contracts/reserve/src/state/config.rs

```diff
@@ -1,29 +1,28 @@
 use serde::{Deserialize, Serialize};
 
-use platform::contract::CodeId;
+use platform::contract::Code;
 use sdk::{
     cosmwasm_std::Storage,
     cw_storage_plus::Item,
     schemars::{self, JsonSchema},
 };
 
-use crate::{api::InstantiateMsg, error::Result};
+use crate::error::Result;
 
-#[derive(Serialize, Deserialize, Clone, Debug, Eq, PartialEq, JsonSchema)]
+#[derive(Serialize, Deserialize, Clone, Eq, PartialEq, JsonSchema)]
 pub struct Config {
-    lease_code_id: CodeId,
+    lease_code: Code,
 }
 
 impl Config {
     const STORAGE: Item<'static, Self> = Item::new("config");
 
-    #[cfg(test)]
-    pub const fn new(lease_code_id: CodeId) -> Self {
-        Self { lease_code_id }
+    pub fn new(lease_code: Code) -> Self {
```

Comment:
When it doesn't cost us anything, it's nicer to have it as `const fn` so we don't have to make massive changes later. ;)
```suggestion
    pub const fn new(lease_code: Code) -> Self {
```